authorization/token/service/token_service.go

```diff
@@ -0,0 +1,249 @@
+package service
+
+import (
+	"context"
+	"github.com/fabric8-services/fabric8-auth/application/service"
+	"github.com/fabric8-services/fabric8-auth/application/service/base"
+	tokenPkg "github.com/fabric8-services/fabric8-auth/authorization/token"
+	tokenRepo "github.com/fabric8-services/fabric8-auth/authorization/token/repository"
+	"github.com/fabric8-services/fabric8-auth/token"
+	"github.com/fabric8-services/fabric8-auth/token/tokencontext"
+
+	"github.com/dgrijalva/jwt-go"
+	servicecontext "github.com/fabric8-services/fabric8-auth/application/service/context"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/satori/go.uuid"
+
+	"github.com/fabric8-services/fabric8-auth/login"
+	"time"
+)
+
+type TokenServiceConfiguration interface {
+	GetRPTTokenMaxPermissions() int
+}
+
+type tokenServiceImpl struct {
+	base.BaseService
+	config TokenServiceConfiguration
+}
+
+func NewTokenService(context servicecontext.ServiceContext, conf TokenServiceConfiguration) service.TokenService {
+	return &tokenServiceImpl{
+		BaseService: base.NewBaseService(context),
+		config:      conf,
+	}
+}
+
+// Audit verifies an existing token in respect to its privileges for a specified resource.  It starts by validating
+// the status of the token passed in the request, and if that token is currently valid and contains the specified
+// resource, returns the same token.  If the token is invalid or outdated, or doesn't contain the specified resource,
+// then a new token is generated and returned.
+// Returns an empty string if no new token has been issued, otherwise returns the new token string
+func (s *tokenServiceImpl) Audit(ctx context.Context, tokenString string, resourceID string) (string, error) {
+
+	// First let's make sure we can load the identity from the context
+	identity, err := login.LoadContextIdentityIfNotDeprovisioned(ctx, s.Repositories())
+	if err != nil {
+		return "", err
+	}
+
+	// Get the token manager from the context
+	tm := tokencontext.ReadTokenManagerFromContext(ctx)
+	if tm == nil {
+		log.Error(ctx, map[string]interface{}{
+			"token": tm,
+		}, "missing token manager")
+
+		return "", errors.NewInternalErrorFromString(ctx, "Missing token manager")
+	}
+	manager := tm.(token.Manager)
+
+	// Now parse the token string that was passed in
+	tokenClaims, err := manager.ParseToken(ctx, tokenString)
+	if err != nil {
+		return "", errors.NewBadParameterErrorFromString("tokenString", tokenString, "invalid token string could not be parsed")
+	}
+
+	// Now that we have the identity and have parsed the token, we can see if we have a record of the token in the database
+	var tokenID uuid.UUID
+
+	// Extract the kid from the token
+	tokenID, err = uuid.FromString(tokenClaims.Id)
+	if err != nil {
+		// TODO Ignore? or perhaps log
```

Comment:
If the ID is not a UUID , then `err` will be not nil. Which implies something is not right. Maybe, return an error then?